flycheck.el

```diff
@@ -3685,7 +3690,16 @@ Intended for use with `next-error-function'."
   (let ((pos (flycheck-next-error-pos n reset)))
     (if pos
         (goto-char pos)
-      (user-error "No more Flycheck errors"))))
+      (let ((result (cl-remove-if
+                     (lambda (err)
+                       (eq (current-buffer)
+                           (flycheck-error-buffer err)))
+                     flycheck-current-errors)))
+        (if result
+            (let ((error (car result)))
+              (flycheck-jump-to-error error)
+              (flycheck-buffer))
+          (user-error "No more Flycheck errors"))))))
```

Comment:
I think this implies that you assume that checking the new buffer will yield the same errors as checking the original buffer did for the new buffer which doesn't hold for C and C++. Checking a header file alone can yield different errors than checking an implementation file including that header file.